internal/util/result.go

```diff
@@ -0,0 +1,19 @@
+package util
+
+type Results struct {
+	Successes int
+	Failures  int
+	Err       error
+}
+
+func (r *Results) Total() int {
+	return r.Successes + r.Failures
+}
+
+func (r *Results) SuccessRate() float64 {
+	return float64(r.Successes) / float64(r.Total())
```

Comment:
Should we protect against division by zero in both the `SuccessRate` and `FailureRate` functions, with something like:

```
if r.Total() == 0 {
  return 0
}
```

In both implementations?

Reply:
Makes sense. I'll add that.